File: examples/quickstart/python/quickstart.py
"""Lucairn quickstart — Python.

Sends a synthetic German medical PII prompt through gateway.lucairn.eu and
prints the redacted response together with the verifiable certificate URL.

Run:
    export LUCAIRN_API_KEY=lcr_live_...
    export ANTHROPIC_API_KEY=sk-ant-...
    pip install -r requirements.txt
    python quickstart.py
"""
from __future__ import annotations

import json
import os
import sys

import httpx

GATEWAY_URL = os.environ.get("LUCAIRN_GATEWAY_URL", "https://gateway.lucairn.eu")
MODEL = os.environ.get("LUCAIRN_MODEL", "claude-sonnet-4-6")

# Synthetic patient note. All identifiers are fake.
PROMPT = (
    "Bitte fasse den folgenden Patientenbefund auf Deutsch in 2 Sätzen zusammen.\n\n"
    "Patientin: Anna Schmidt, geboren 12.03.1985, "
    "IBAN DE89 3704 0044 0532 0130 00.\n"
    "Diagnose: Hypertonie. Empfehlung: Tägliche Blutdruck-Messung, "
    "Reduktion Salzkonsum, Beta-Blocker 50mg."
)


def main() -> int:
    lucairn_key = os.environ.get("LUCAIRN_API_KEY")
    anthropic_key = os.environ.get("ANTHROPIC_API_KEY")
    if not lucairn_key or not anthropic_key:
        sys.stderr.write(
            "ERROR: set LUCAIRN_API_KEY and ANTHROPIC_API_KEY env vars before "
            "running.\n"
        )
        return 1

    headers = {
        "x-api-key": lucairn_key,
        "X-Upstream-Key": anthropic_key,
        "content-type": "application/json",
    }
    body = {
        "model": MODEL,
        "max_tokens": 256,
        "messages": [{"role": "user", "content": PROMPT}],
    }

    response = httpx.post(
        f"{GATEWAY_URL}/v1/messages",
        headers=headers,
        json=body,
        timeout=60.0,
    )
    if response.status_code != 200:
        sys.stderr.write(
            f"Gateway returned {response.status_code}: {response.text}\n"
        )
        return 1

    payload = response.json()
    content_blocks = payload.get("content", [])
    text = "\n".join(b.get("text", "") for b in content_blocks if b.get("type") == "text")
    compliance = (payload.get("metadata") or {}).get("dsa_compliance", {})

    print("── LLM response ────────────────────────────────────────")
    print(text or "(no text content)")
    print()
    print("── Compliance metadata ─────────────────────────────────")
    print(json.dumps(compliance, indent=2, ensure_ascii=False))
    print()
    cert_url = compliance.get("veil_certificate_url")
    if cert_url:
        request_id = cert_url.rsplit("/", 1)[-1]
        verify_url = f"https://lucairn.eu/verify?cert={request_id}"
        print("── Verify the cert ─────────────────────────────────────")
        print(f"Open: {verify_url}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/quickstart/typescript/quickstart.ts
/**
 * Lucairn quickstart — TypeScript / Node 20+.
 *
 * Sends a synthetic German medical PII prompt through gateway.lucairn.eu and
 * prints the redacted response together with the verifiable certificate URL.
 *
 * Run:
 *   export LUCAIRN_API_KEY=lcr_live_...
 *   export ANTHROPIC_API_KEY=sk-ant-...
 *   pnpm install
 *   pnpm exec tsc && node dist/quickstart.js
 *
 * Uses the built-in fetch (Node >= 18). The published `@lucairn/sdk` package
 * is not yet feature-complete for /v1/messages; raw fetch is the supported
 * path today and is exactly what the SDK will wrap when it ships.
 */

const GATEWAY_URL = process.env.LUCAIRN_GATEWAY_URL ?? "https://gateway.lucairn.eu";
const MODEL = process.env.LUCAIRN_MODEL ?? "claude-sonnet-4-6";

// Synthetic patient note. All identifiers are fake.
const PROMPT =
  "Bitte fasse den folgenden Patientenbefund auf Deutsch in 2 Sätzen zusammen.\n\n" +
  "Patientin: Anna Schmidt, geboren 12.03.1985, " +
  "IBAN DE89 3704 0044 0532 0130 00.\n" +
  "Diagnose: Hypertonie. Empfehlung: Tägliche Blutdruck-Messung, " +
  "Reduktion Salzkonsum, Beta-Blocker 50mg.";

interface ContentBlock {
  type: string;
  text?: string;
}

interface DSACompliance {
  request_id: string;
  veil_certificate_url?: string;
  veil_summary_url?: string;
  pii_in_ai?: boolean;
  identity_in_ai?: boolean;
  sanitizer_layers?: string[];
  redaction_count: number;
  latency_ms: number;
}

interface AnthropicResponse {
  content?: ContentBlock[];
  metadata?: { dsa_compliance?: DSACompliance };
}

async function main(): Promise<number> {
  const lucairnKey = process.env.LUCAIRN_API_KEY;
  const anthropicKey = process.env.ANTHROPIC_API_KEY;
  if (!lucairnKey || !anthropicKey) {
    process.stderr.write(
      "ERROR: set LUCAIRN_API_KEY and ANTHROPIC_API_KEY env vars before running.\n",
    );
    return 1;
  }

  const response = await fetch(`${GATEWAY_URL}/v1/messages`, {
    method: "POST",
    headers: {
      "x-api-key": lucairnKey,
      "X-Upstream-Key": anthropicKey,
      "content-type": "application/json",
    },
    body: JSON.stringify({
      model: MODEL,
      max_tokens: 256,
      messages: [{ role: "user", content: PROMPT }],
    }),
  });

  if (!response.ok) {
    const text = await response.text();
    process.stderr.write(`Gateway returned ${response.status}: ${text}\n`);
    return 1;
  }

  const payload = (await response.json()) as AnthropicResponse;
  const text = (payload.content ?? [])
    .filter((b) => b.type === "text" && typeof b.text === "string")
    .map((b) => b.text!)
    .join("\n");
  const compliance = payload.metadata?.dsa_compliance;

  console.log("── LLM response ────────────────────────────────────────");
  console.log(text || "(no text content)");
  console.log();
  console.log("── Compliance metadata ─────────────────────────────────");
  console.log(JSON.stringify(compliance ?? {}, null, 2));
  console.log();
  if (compliance?.veil_certificate_url) {
    const requestId = compliance.veil_certificate_url.split("/").pop() ?? "";
    console.log("── Verify the cert ─────────────────────────────────────");
    console.log(`Open: https://lucairn.eu/verify?cert=${requestId}`);
  }
  return 0;
}

main().then(
  (code) => process.exit(code),
  (err) => {
    process.stderr.write(`unhandled error: ${err}\n`);
    process.exit(1);
  },
);
